__tests__/integration.js

```diff
@@ -70,3 +72,43 @@ test('--mutex network', async () => {
     execa(command, ['add', 'foo'].concat(args), options),
   ]);
 });
+
+test('cache folder fallback', async () => {
+  const cwd = await makeTemp();
+  const cacheFolder = path.join(cwd, '.cache');
+
+  await fs.mkdirp(cacheFolder);
+
+  const command = path.resolve(__dirname, '../bin/yarn');
+  const args = ['--preferred-cache-folder', cacheFolder];
+
+  const options = {cwd};
+
+  {
```

Comment:
I'd prefer if you can make this a function since it would make the code easier to follow. I don't remember seen usage unnamed, custom blocks in a project I have worked on and it would really surprise me as it did right now. I understand that we can leverage the block-scoping with `let` and `const` but I think this makes the code less maintainable and harder to reason about.